src/util/change-types.ts

```diff
@@ -310,3 +310,57 @@ export const increaseZIndexInBatch = (
         graph.setEdgeAttribute(s, 'zIndex', z + value);
     });
 };
+
+const changeStationIntType = (
+    graph: MultiDirectedGraph<NodeAttributes, EdgeAttributes, GraphAttributes>,
+    station: StnId,
+    to: 'int' | 'basic'
+) => {
+    const getDestinationType = (type: string) => {
+        if (type.endsWith('-int') && to === 'basic') {
+            return type.replace(/-int$/, '-basic');
+        } else if (type.endsWith('-basic') && to === 'int') {
+            return type.replace(/-basic$/, '-int');
+        }
+        return undefined;
+    };
+
+    const currType = graph.getNodeAttribute(station, 'type') as string;
+    const destType = getDestinationType(currType);
+
+    if (!destType) return;
+    if (!Object.values(StationType).includes(destType as StationType)) {
+        return;
+    }
+
+    changeStationType(graph, station, destType as StationType);
+};
+
+export const autoChangeStationIntType = (
```

Comment:
This name is vague, as it does not state under what circumstances this function will work. Rename it to something like `checkStationInt` and add a comment better explaining when it should be called and what it will do.

Also I think `to` is not needed as you could know it in this function.